File: 389-find-the-difference/find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        char extraChar = 0;
        for(int i=0; i<s.length(); i++){
            extraChar = (char) (extraChar ^ s.charAt(i)); // XOR 
        }
        for(int i=0; i<t.length(); i++){
            extraChar = (char) (extraChar ^ t.charAt(i));
        }
    return extraChar;
    }
}

File: 389-find-the-difference/find-the-difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        char extraChar = 0;
        for(char i : s){
            extraChar = extraChar ^ i; // XOR 
        }
        for(char i : t){
            extraChar = extraChar ^ i;
        }
    return extraChar;
    }
};